src/strat_engine/blockdev.rs

```diff
@@ -381,4 +381,33 @@ impl BlockDev {
     pub fn dstr(&self) -> String {
         format!("{}:{}", self.dev.major, self.dev.minor)
     }
+
+    /// List all used areas in the blockdev.
+    fn used_areas(&self) -> Vec<(SectorOffset, Sectors)> {
+        let mut used = Vec::new();
+
+        // Flag start mda zone as used
+        used.push((SectorOffset(0),
+                   BDA_STATIC_HDR_SIZE + self.mda_sectors + self.reserved_sectors));
+        // Flag end mda zone as used
+        used.push((SectorOffset(*self.sectors - *BDA_STATIC_HDR_SIZE - *self.mda_sectors),
+                   BDA_STATIC_HDR_SIZE + self.mda_sectors));
+        used
+    }
+
+    /// List all unused, available extents in this blockdev.
+    pub fn avail_areas(&self) -> Vec<(SectorOffset, Sectors)> {
+        let used = self.used_areas();
+        assert_eq!(used.len(), 2);
```

Comment:
Is this correct? The code below seems extremely ornate if the number of these "reserved" areas is certainly just two.